### DISCUSSION TITLE
Users in multiple groups

### DISCUSSION THREAD

**Discussion by jalouke** [mealie-recipes/mealie#824] [Feature Request]
I want to start off by saying that this recipe manager is super cool, and I am really loving what you have created. I would love to help out and contribute, but I would say that I am a novice when it comes to Python. and I don't have any experience when it comes to the scale of application development.

I have a feature suggestion that relates to user groups. I think it would be nice if users could be in multiple groups to share meal plans. An expansion on that is to be able to share a meal with specific people for a party or event.

**Comment by hay-kot** [comment]
Please use the feature request template.

https://github.com/hay-kot/mealie/issues/317

**Comment by Ravio1i** [comment]
I'm no the original requester, but I woul like to create the feature request

**Before submitting this feature request I have**
_COMPLETE AND FILL THESE OUT_
- [X] Reviewed the documentation and verified that my feature request isn't already a feature.
- [X] Reviewed the existing feature requests and my feature has not been requested.
- [x] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
Currently one user is attached to ony group. However, there are scenarios, were one user should be part of multiple groups.
E.g I have the scenario that I have my personal group where I would like to create and track my recipes, shopping list and week plan. But I have another group with friends were there is a week plan for 2 days a week, aswell. Currently I can only achieve this by creating another user which is part of the shared group. 

Additionally,I would like to share my personal recipes and vice versa from and to the shared group. Currently I can only achieve this by making a recipe public and sending the link or duplicating recipes.

Lastly, but only nice-to-have is a unified week planner: This means that I can add my recipes for the week to either my personal group or the shared group.

**(Optional): Suggest A Solution**
* [ ] User can be member of multiple groups
* [ ] Recipes are present among multiple groups
* [ ] (Nice-to-have) Unified recipe week planner

**Comment by dmick92** [comment]
Is this being considered?

**Reply by Kuchenpirat** [reply]
I don't think we will be implementing this. Groups are designed to be pretty much isolated instances. 

Does your usecase for this feature differ from the situation above?
Always good to have more info on why a feature is needed.

**Reply by dmick92** [reply]
I've got it set up for my family and it's fine, but my wife wants to create an instance for her and her friends that's isolated from the main home group. If I create a new group it'll work fine for her friends but she'll be unable to access it as she's part of the home group.

In essence I'm hosting 1 instance for multiple households. each household should be it's own group. This is how Mealie appears to be built. But 1 household, my parents, they have a family cookbook that's been passed down for generations and has been added to their group. Ideally they could give access to my household(group) in order to share the recipes with their family.

**Reply by Kuchenpirat** [reply]
Take a look at #2134 it might fullfill your needs and we are way more likely to implement this compared to removing the group isolation. 

The idea is that Recipes, Foods etc would be shared across all group members, but meal plans, shopping lists etc are seperate.

**Reply by dmick92** [reply]
Thanks for that, it looks like it would fulfill my needs. However there doesn't appear to be much progress/updates on it considering the positive feedback.

**Reply by Kuchenpirat** [reply]
It would require a considerate amount of work in many different components which is the reason nobody has gotten arround to it yet.

But yes it is one of our most requested features and one that does make a lot of sense 👍
